service/controller/clusterapi/v30/resource/s3object/desired.go

```diff
@@ -24,13 +29,24 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 		return nil, microerror.Mask(err)
 	}
 
+	var cluster v1alpha1.Cluster
 	var clusterCerts certs.Cluster
 	var clusterKeys randomkeys.Cluster
 	{
 		g := &errgroup.Group{}
 
 		g.Go(func() error {
-			certs, err := r.certsSearcher.SearchCluster(key.ClusterID(&cr))
+			m, err := r.cmaClient.ClusterV1alpha1().Clusters(cr.GetNamespace()).Get(key.ClusterID(cr), metav1.GetOptions{})
+			if err != nil {
+				return microerror.Mask(err)
+			}
+			cluster = *m
```

Comment:
The most straight forward thing to do is to fetch the `Cluster` CR and use it below. This is because different controllers execute this resource so sometimes it is the `Cluster` but sometimes it is the `MachineDeployment` and I think before we start using any weird conditionals in the resources we should have separate resources. Since we already fetch a couple of information in parallel here this should be fine for now.